types/node/index.d.ts

```diff
@@ -780,101 +787,60 @@ declare module "http" {
         socketPath?: string;
         method?: string;
         path?: string;
-        headers?: { [key: string]: any };
+        headers?: OutgoingHeaders;
         auth?: string;
         agent?: Agent | boolean;
         timeout?: number;
     }
 
-    export interface Server extends net.Server {
+    export class Server extends net.Server {
+        constructor(req: IncomingMessage, res: ServerResponse);
         setTimeout(msecs: number, callback: Function): void;
         maxHeadersCount: number;
         timeout: number;
         listening: boolean;
     }
-    /**
-     * @deprecated Use IncomingMessage
-     */
-    export interface ServerRequest extends IncomingMessage {
-        connection: net.Socket;
-    }
-
-    export interface ServerResponseHeaders {
-      [key: string]: number | string | string[];
-    }
-
-    export interface ServerResponse extends stream.Writable {
-        // Extended base methods
-        write(buffer: Buffer): boolean;
-        write(buffer: Buffer, cb?: Function): boolean;
-        write(str: string, cb?: Function): boolean;
-        write(str: string, encoding?: string, cb?: Function): boolean;
-        write(str: string, encoding?: string, fd?: string): boolean;
 
-        addTrailers(headers: ServerResponseHeaders): void;
+    export class OutgoingMessage extends stream.Writable {
```

Comment:
Technically it inherits from stream internally but all the types match up with a writable stream.

![image](https://user-images.githubusercontent.com/1088987/27835822-a66b177e-60aa-11e7-8a89-0163f1f8455c.png)